ipc/nano/src/lib.rs

```diff
@@ -54,9 +54,9 @@ impl<S> GuardedSocket<S> where S: WithSocket<Socket> {
 }
 
 impl<S> Deref for GuardedSocket<S> where S: WithSocket<Socket> {
-    type Target = S;
+    type Target = Arc<S>;
 
-    fn deref(&self) -> &S {
+    fn deref(&self) -> &Arc<S> {
```

Comment:
i am becoming wary of anything that uses `&Arc<T>` since it makes it too easy to perform unnecessary atomic operations.

why not have an explicit `clone_socket` method?